examples/z_queryable.c

```diff
@@ -26,19 +26,19 @@ const char *value = "Queryable from C!";
 z_keyexpr_t keyexpr;
 
 void query_handler(const z_query_t *query, void *context) {
-    char *keystr = z_keyexpr_to_string(z_query_keyexpr(query));
+    z_owned_str_t keystr = z_keyexpr_to_string(z_query_keyexpr(query));
     z_bytes_t pred = z_query_parameters(query);
     z_value_t payload_value = z_query_value(query);
     if (payload_value.payload.len > 0) {
-        printf(">> [Queryable ] Received Query '%s?%.*s' with value '%.*s'\n", keystr, (int)pred.len, pred.start,
-            (int)payload_value.payload.len, payload_value.payload.start);
+        printf(">> [Queryable ] Received Query '%s?%.*s' with value '%.*s'\n", z_loan(keystr), (int)pred.len,
+               pred.start, (int)payload_value.payload.len, payload_value.payload.start);
     } else {
-        printf(">> [Queryable ] Received Query '%s?%.*s'\n", keystr, (int)pred.len, pred.start);
+        printf(">> [Queryable ] Received Query '%s?%.*s'\n", z_loan(keystr), (int)pred.len, pred.start);
     }
     z_query_reply_options_t options = z_query_reply_options_default();
     options.encoding = z_encoding(Z_ENCODING_PREFIX_TEXT_PLAIN, NULL);
     z_query_reply(query, z_keyexpr((const char *)context), (const unsigned char *)value, strlen(value), &options);
-    free(keystr);
+    z_drop(&keystr);
```

Comment:
In Zenoh-Pico, we have it implemented as `z_drop(z_move(keystr))`.
I know it might seem redundant but by doing so we dont break the rule of being explicit when moving things to the Zenoh API.